enumerable.rb

```diff
@@ -0,0 +1,159 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    arr = [Hash, Range].member?(self.class) ? to_a.flatten : self
+    i = 0
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = [Hash, Range].member?(self.class) ? to_a.flatten : self
+    i = 0
+    while i < arr.length
+      yield(arr[i], i) if arr[i].is_a? Integer
+      i += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = [Hash, Range].member?(self.class) ? to_a.flatten : self
+    results = []
+    arr.my_each { |x| results << x if yield(x) }
+    results
+  end
+
+  def my_all?(test = nil)
+    check = true
+    if block_given?
+      my_each { |x| check = false unless yield(x) }
+      check
+    elsif test.is_a? Class
+      my_each { |x| check = false if x.class != test }
+    elsif test.is_a? Regexp
+      my_each { |x| check = false unless test.match(x) }
+    else
+      t_ar = [nil, false]
+      my_each { |x| check = false if t_ar.include?(x) || x != test }
+    end
+    check
+  end
+
+  def my_any?(test = nil)
+    check = false
+    if block_given?
+      my_each { |x| check = true if yield(x) }
+    elsif test.is_a? Class
+      my_each { |x| check = true if x.class == test }
+    elsif test.is_a? Regexp
+      my_each { |x| check = true if test.match(x) }
+    else
+      test_array = [true, [], {}]
+      my_each { |x| check = true if test_array.include?(x) || include?(test) }
+    end
+    check
+  end
```

Comment:
Same problems from my_all? Maybe they are working the other-way around? Im getting false when expecting true.